bootstrap/bootstrap.go

```diff
@@ -1437,6 +1437,23 @@ func (b *Bootstrap) defaultCommandPhase() error {
 		return runDeprecatedDockerIntegration(b.shell, []string{cmdToExec})
 	}
 
+	if sensitiveVarNames, ok := b.shell.Env.Get("BUILDKITE_REDACTED_VARS"); ok {
```

Comment:
Because this is looking at the current shell environment, it means it can be overridden at a build or pipeline level. I'd say that this is probably not desirable.

This probably needs to be an AgentConfiguration that can only be set in the agent configuration or in a hook. I can help with this bit if you need @dbaggerman.

Reply:
If you can point me in the right direction for this, that would be great.